### ISSUE TITLE
Backoff GPS checking once GPS signal lost for a time

### ISSUE THREAD

**Issue by ehem** [zamojski/TowerCollector#24]
If GPS signal is lost and TowerCollector is running it appears TowerCollector starts to eat the battery.  I'm guessing TowerCollector continuously tries to regain location information.  Checking frequently immediately after the GPS signal has been lost is okay, but once signal has been lost for a period TowerCollector should back off and check less frequently to save battery.  Most likely if minimal motion is seen, the person/phone is inside a building; larger motion might suggest a tunnel.

**Comment by zamojski** [comment]
I think this is duplicated with #6. If you have different opinion please let me know.

**Comment by ehem** [comment]
No, even though there is substantial overlap of functionality this _is_ a distinct issue.  You might well fix issue #6 without fixing this, or you could fix #24 without fixing #6.

Once a device has lost GPS fix, I imagine TowerCollector either immediately retries for a GPS fix or else waits perhaps 1 second.  This is okay for a time, but once GPS signal has been lost for a time, TowerCollector needs to increase the delay for retrying for a GPS fix.  If motion sensors on a device indicate it is traveling at less than 2 m/s the maximum retry delay might be 300-600 seconds, if a device is moving 10+ m/s the maximum retry delay might be 60 seconds.

This can happen if you're in an internal office of a building which blocks the GPS signal.  This can also happen if you enter the Chunnel.

For #24, having motion sensors is allows optimizing the maximum delay (and how quickly the delay increases), but the observation is continuous periods of GPS signal loss do happen.  For #6 the observation is, if there is **no** motion there isn't a need to check GPS at all.

**Comment by Lee-Carre** [comment]
I habitually enable Tower Collector (and similar) while I'm out & about. Sometimes I won't need my phone for anything else, so it remains in my pocket (connected to an external battery).

During the course of the day, I'm often inside buildings for periods of time, without telling Tower Collector to stop.

So, having Tower Collector automatically handle those situations, would help.
